packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -41,7 +42,7 @@ interface ComboboxContentProps {
    * @optional
    * @type string
    */
-  readonly selected?: string | number;
+  readonly selected?: string | number | null;
```

Comment:
how do we feel about the whole `string | number | null` type for `selected` vs `ComboboxOption | null`?

or maybe if there's a TS way to accept excess properties so we need the minimum of `label` and `id` but don't care if you give us more? then we don't need to specifically use `ComboboxOption` but I digress...

the reason I ask is because we're a bit mismatched with the input vs output now where we have

`readonly onSelect: (selection: ComboboxOption) => void;`

so the incoming value is just a number or string, but the value we get back will be `{ id: 1, label: "whatever" }` meaning you'd need a setup like this

```
const [ selected, setSelected ] = useState<number | null>(1);

<Combobox>
  <Combobox.Content
   onSelect={selection => setSelected(selection.id)}
   selected={selected}
   />
</Combobox>
```

where it's that `selection.id` that feels a bit weird to me. I think there are pros and cons to either approach, but whichever we choose let's make the input and output match!

Reply:
ok ya good call they should definitely be the same.

Reply:
after playing around with this, I feel like `ComboboxOption | null` is preferable as the type for `selected` which then matches the output we currently have

with `selected` as `number | string | null` and applying those same types to the corresponding outgoing `selection` you get from the `onSelect` callback, it becomes a bit inconvenient

say your selected state is `const [ hat, setHat ] = useState<number | null>(2)`

trying to do this will have a problem `onSelect={selection => setHat(selection)}`. we've said `hat` can be either a number or null. however; `selection` can be a number, null or a string. it's that third possibility that causes the issue. you can do some type coercion but that doesn't feel very elegant to me

whereas if we use `ComboboxOption` you can simply pass in a `selected` and `setHat(selection)` with no other steps required. it's very similar to how we handle the `onChange` of `Autocomplete` which uses an `Option`